lib/config.js

```diff
@@ -13,15 +13,15 @@ module.exports.settings = function () {
   var file_settings = atom.workspace.getActiveTextEditor().getPath() + SETTINGS_FILENAME;
   var directory_settings = path.join(path.dirname(file_settings), SETTINGS_FILENAME);
   var config_file = "";
-
+  var project_id = require("./utility").guessProjectId();
   if (fs.existsSync(file_settings)) {
       config_file = file_settings;
   } else if (fs.existsSync(directory_settings)) {
       config_file = directory_settings;
   }
 
-  if (config_file == "" && atom.project.getPaths()[0] != undefined) {
-      var project_path = atom.project.getPaths()[0];
+  if (config_file == "" && atom.project.getPaths()[project_id] != undefined) {
+      var project_path = atom.project.getPaths()[project_id];
```

Comment:
You can simplify this whole thing with this code:

```js
const projectPath = atom.project.relativizePath(filePath)[0];
```

Note that you will need to check for a `null` response, so most often the following code is used to fall back to the directory of the current file if it hasn't been opened inside of a project:

```js
let projectPath = atom.project.relativizePath(filePath)[0];
if (projectPath === null) {
  projectPath = path.dirname(filePath);
}
```

This _might_ require some refactoring as having this in the settings is a bit... odd. Normally things dependent on the path of the file are handled in the `lint()` function as it is passed the editor of the file you are linting so there is no guesswork as to which editor you should be working with.